app/components/check_your_answers_summary_component.rb

```diff
@@ -10,11 +10,70 @@ def initialize(model:, title:, fields:, delete_link_to: nil)
   attr_reader :title
 
   def rows
-    fields.map { |key, opts| opts ? row_for_field(key, opts) : nil }.compact
+    fields_with_translations.filter_map do |field|
+      field.options ? row_for_field(field.key, field.options) : nil
+    end
   end
 
   private
 
+  class Field
+    attr_reader :key, :options
+
+    def initialize(field_array)
+      @key = field_array[0]
+      @options = field_array[1]
+    end
+  end
```

Comment:
I wonder if there's much benefit to having this class. I can see we needed a way of retaining the reference to the original key, but I wonder if it's better to model it as a hash with an extra field for the title, so something like this:

```rb
# original field
{
  key: "document",
}

# translated field
{
  key: "document",
  title: "document translations",
  translations: true
}
```

I think it would simplify some of this code, remove the need for `fields_as_array` and `translation_row_for`, and it's got the added benefit that we can support an optional `title` field for all the field types rather than just translated documents.

Getting the title of a field would have to change to:

```rb
opts.fetch(:title, opts.fetch(:key, key.to_s.humanize)))
```

I notice at the moment we've got a new method `row_title_for` in this PR, but we only use it for translations, whereas we could use it for all the fields as well if we wanted to.

Just my opinion though, I'm happy to keep the current implement to move forward.

Reply:
There is a downside to this which is that everywhere we currently write `key: ...` when using the `CheckYourAnswersSummaryComponent` we'd have to change it to `title: ...` but actually I think this makes more sense than `key` anyway.

Reply:
The `Field` class could be a Hash as you suggest. We refactored it to that as originally we were passing the array around all over the place and accessing it with [index] and it was horrible. I'll have a look but object vs hash here is probably just a stylistic difference and we could do the other things with either. I think we still need `fields_as_array` as `fields` is a hash originally but we need to position the new rows after the translatable row which was simpler to do using the index of the array.

Reply:
Yeah you're right about the `fields_as_array`, I sneakily edited my comment after writing it 😅

I think I still prefer my suggestion as it allows us to be consistent about the `title` and `key` and treat it the same for all types of rows, but I'm happy for you to ignore this suggestion too.